lib/services/devices/deviceService.js

```diff
@@ -205,45 +205,97 @@ function registerDevice(deviceObj, callback) {
         return attribute;
     }
 
-    function prepareDeviceData(deviceObj, callback) {
-        var deviceData = _.clone(deviceObj),
-            fields = ['service', 'subservice', 'lazy', 'internalAttributes',
-                'active', 'staticAttributes', 'commands', 'subscriptions'],
-            defaults = [null, null, [], [], [], [], []];
+    function checkDuplicates(deviceObj, innerCb) {
+        config.getRegistry().get(deviceObj.id, deviceObj.service, deviceObj.subservice, function(error, device) {
+            if (!error) {
+                innerCb(new errors.DuplicateDeviceId(deviceObj.id));
+            } else {
+                innerCb();
+            }
+        });
+    }
 
-        if (!deviceData.type) {
-            deviceData.type = config.getConfig().defaultType;
+    function findConfigurationGroup(deviceObj, callback) {
+        function handlerGroupFind(error, group) {
+            if (error || !group) {
+                callback(null, null);
+            } else {
+                callback(null, group);
+            }
```

Comment:
The whole previous `if`clause seems to be "equivalent" to `callback(null, group)` (the only difference being that the second argument of the `callback` will be passed as `undefined` instead as `null`.